internal/test/smoke/size_limits_test.go

```diff
@@ -0,0 +1,250 @@
+package smoke
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/labstack/echo/v4"
+	"github.com/podengo-project/idmsvc-backend/internal/api/public"
+	"github.com/podengo-project/idmsvc-backend/internal/config"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type sizeLimitConfig struct {
+	IdleTimeout     time.Duration
+	ReadTimeout     time.Duration
+	WriteTimeout    time.Duration
+	HeaderSizeLimit int
+	BodySizeLimit   int
+}
+
+// SuiteTokenCreate is the suite token for smoke tests at /api/idmsvc/v1/domains/token
+type SuiteLimitSize struct {
+	SizeLimitConfig map[string]sizeLimitConfig
+	SuiteBase
+}
+
+func generateConfig(customSizeLimitConfig sizeLimitConfig) *config.Config {
+	cfg := &config.Config{}
+	_ = config.Load(cfg)
+	if customSizeLimitConfig.IdleTimeout != 0 {
+		cfg.Application.IdleTimeout = customSizeLimitConfig.IdleTimeout
+	}
+	if customSizeLimitConfig.ReadTimeout != 0 {
+		cfg.Application.ReadTimeout = customSizeLimitConfig.ReadTimeout
+	}
+	if customSizeLimitConfig.WriteTimeout != 0 {
+		cfg.Application.WriteTimeout = customSizeLimitConfig.WriteTimeout
+	}
+	if customSizeLimitConfig.HeaderSizeLimit != 0 {
+		cfg.Application.SizeLimitRequestHeader = customSizeLimitConfig.HeaderSizeLimit
+	}
+	if customSizeLimitConfig.BodySizeLimit != 0 {
+		cfg.Application.SizeLimitRequestBody = customSizeLimitConfig.BodySizeLimit
+	}
+	return cfg
+}
+
+func (s *SuiteLimitSize) SetupTest() {
+	testName := s.T().Name()
+	s.SuiteBase.Config = generateConfig(s.SizeLimitConfig[testName])
+	s.SuiteBase.SetupTest()
+}
+
+func (s *SuiteLimitSize) TearDownTest() {
+	s.SuiteBase.TearDownTest()
+}
+
+func (s *SuiteLimitSize) TestReadTimeout() {
+	t := s.T()
+	requestID := "test_read_timeout_with_create_token"
+
+	// Use a pipe that force the timeout
+	pr, pw := io.Pipe()
+	go func() {
+		defer pw.Close()
+		// Wait a time bigger than the ReadTimeout
+		time.Sleep(s.Config.Application.ReadTimeout + 1*time.Second)
+	}()
+
+	// Create a HTTP client and sent the request
+	client := &http.Client{}
+
+	hdr := http.Header{}
+	url := s.DefaultPublicBaseURL() + "/domains/token"
+	s.addRequestID(&hdr, requestID)
+	req, err := http.NewRequestWithContext(context.Background(), http.MethodPost, url, pr)
+	require.NoError(t, err)
+	require.NotNil(t, req)
+	req.Header.Set(echo.HeaderContentType, echo.MIMEApplicationJSON)
+	for key, value := range hdr {
```

Comment:
I'm looking at the code and I don't understand much the purpose of this for loop.  E.g. when I tried:
```diff
-       hdr := http.Header{}
        url := s.DefaultPublicBaseURL() + "/domains/token"
-       s.addRequestID(&hdr, requestID)
        req, err := http.NewRequestWithContext(context.Background(), http.MethodPost, url, pr)
        require.NoError(t, err)
        require.NotNil(t, req)
        req.Header.Set(echo.HeaderContentType, echo.MIMEApplicationJSON)
-       for key, value := range hdr {
-               req.Header.Set(key, strings.Join(value, "; "))
-       }
        // Override
        s.currentXRHID = &s.userXRHID
        s.addXRHIDHeader(&req.Header, s.currentXRHID)
+       s.addRequestID(&req.Header, requestID)
```

then the test worked as well and it seems more readable to me.